pkg/network/protocols/http/shared_libraries.go

```diff
@@ -105,65 +103,52 @@ type soWatcher struct {
 type soRegistry struct {
 	m     sync.Mutex
 	byID  map[pathIdentifier]*soRegistration
-	byPID map[uint32]*soRegistration
+	byPID map[uint32]map[pathIdentifier]struct{}
 
-	// if we can't register a uprobe we don't try more than once
+	// if we can't register an uprobe we don't try more than once
 	blocklistByID map[pathIdentifier]struct{}
 }
 
 func newSOWatcher(perfHandler *ddebpf.PerfHandler, rules ...soRule) *soWatcher {
-	allFilters := make([]string, len(rules))
-	for i, r := range rules {
-		allFilters[i] = r.re.String()
-	}
-
-	all := regexp.MustCompile(fmt.Sprintf("(%s)", strings.Join(allFilters, "|")))
 	return &soWatcher{
 		procRoot:       util.GetProcRoot(),
-		all:            all,
 		rules:          rules,
 		loadEvents:     perfHandler,
 		processMonitor: monitor.GetProcessMonitor(),
 		registry: &soRegistry{
 			byID:          make(map[pathIdentifier]*soRegistration),
-			byPID:         make(map[uint32]*soRegistration),
+			byPID:         make(map[uint32]map[pathIdentifier]struct{}),
 			blocklistByID: make(map[pathIdentifier]struct{}),
 		},
 	}
 }
 
 type soRegistration struct {
-	pathID       pathIdentifier
-	refcount     int
-	unregisterCB func(pathIdentifier) error
+	uniqueProcessesCount int
+	unregisterCB         func(pathIdentifier) error
 }
 
-// Unregister return true if there are no more reference to this registration
-func (r *soRegistration) Unregister() bool {
-	r.refcount--
-	if r.refcount > 0 {
+// unregister return true if there are no more reference to this registration
+func (r *soRegistration) unregister(pathID pathIdentifier) bool {
+	r.uniqueProcessesCount--
+	if r.uniqueProcessesCount > 0 {
 		return false
 	}
 	if r.unregisterCB != nil {
-		if err := r.unregisterCB(r.pathID); err != nil {
-			log.Debugf("error while unregistering %s : %s", r.pathID.String(), err)
+		if err := r.unregisterCB(pathID); err != nil {
+			log.Debugf("error while unregistering %s : %s", pathID.String(), err)
```

Comment:
do we still want to return true here? if yes, please add an explicit comment explaining why

Reply:
that's best effort, if we failed to unregister, we have nothing to do to overcome that